backend/amd64/emit.mlp

```diff
@@ -790,6 +803,21 @@ let emit_instr fallthrough i =
         def_label label;
         I.lea (mem64 NONE 8 R15) (res i 0)
       end
+  | Lop(Ialloc { bytes = n; dbginfo=_; mode = Alloc_local }) ->
+      let r = res i 0 in
+      I.mov (domain_field Domainstate.Domain_local_sp) r;
+      I.sub (int n) r;
+      I.mov r (domain_field Domainstate.Domain_local_sp);
+      I.cmp (domain_field Domainstate.Domain_local_limit) r;
+      let lbl_call = new_label () in
+      I.j L (label lbl_call);
+      let lbl_after_alloc = new_label () in
+      def_label lbl_after_alloc;
+      I.add (domain_field Domainstate.Domain_local_top) r;
```

Comment:
Can you please explain lines 816-817 - what is `Domain_local_top`? Is like 817 for the header?

Reply:
Domain_local_sp is an integer offset which must be added to Domain_local_top to get the actual pointer. Doing it this way means that the value returned by Ibeginregion is an integer that need not be rewritten if the local stack reallocates and moves.

The addition of 8 on line 817 is indeed for the header. The same is done in the existing allocation code (where it's written `I.lea (mem64 NONE 8 R15) (res i 0)`).

Reply:
Make senses, so `Domain_local_sp <= 0`.